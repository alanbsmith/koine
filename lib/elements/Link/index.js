import styled from 'styled-components';
import { Link as UnstyledLink } from 'react-router-dom';
import { applyStyleModifiers } from 'styled-components-modifiers';

import { fontStyles, fontWeights } from '../../modifiers';
import gridScale from '../../utils/gridScale';

const modifierConfig = {
  ...fontStyles,
  ...fontWeights,
};

const Link = styled(UnstyledLink)`
  color: ${props => props.theme.colors.base.link};
  font-size: ${props => gridScale(props, 1.5)};
  font-weight: ${props => props.theme.fontWeights.light};
  text-transform: uppercase;

  &:hover,
  &:focus {
    box-shadow: none;
    color: ${props => props.theme.colors.base.linkHover};
    text-decoration: underline;
  }

  ${applyStyleModifiers(modifierConfig)}
`;

export default Link;
