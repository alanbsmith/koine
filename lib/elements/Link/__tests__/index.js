import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import Link from '../index';
import 'jest-styled-components';

import { renderWithTheme } from '__tests__/helpers';

describe('Link Element', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(
      <Router>
        <Link to="/">Link</Link>
      </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
