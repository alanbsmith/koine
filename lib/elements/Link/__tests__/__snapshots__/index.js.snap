// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Element renders correctly 1`] = `
.c0 {
  color: #1FB6FF;
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

.c0:hover,
.c0:focus {
  box-shadow: none;
  color: #007cb8;
  text-decoration: underline;
}

<a
  className="c0"
  href="/"
  onClick={[Function]}
>
  Link
</a>
`;
